data_structures/src/superblock.rs

```diff
@@ -257,11 +264,41 @@ impl SuperBlockState {
         self.received_superblocks.contains(sbv)
     }
 
-    /// Updates the current superblock signing committee for a given superblock index
-    /// #FIXME This function should be update with issue 1395
-    pub fn update_superblock_signing_committee(&mut self, _current_index: u32) {
-        self.current_signing_committee = self.previous_ars_identities.clone();
+    /// Updates the current superblock signing committee for a given superblock hash
+    pub fn update_superblock_signing_committee(
+        &mut self,
+        _signing_committee_size: u32,
+    ) -> Option<HashSet<PublicKeyHash>> {
+        // If the number of identities is lower than committee_size all the members of the ARS sign the superblock
+        let ars_ordered = &self.previous_ordered_ars_identities;
+        if ars_ordered.len() < usize::try_from(_signing_committee_size).unwrap() {
+            self.current_signing_committee = self.previous_ars_identities.clone();
+            self.current_signing_committee.clone()
+        } else {
+            // Get the number of subsets of _signing_committee_size members
+            let n = ars_ordered.len() as u64 / u64::from(_signing_committee_size);
+            // Start counting the members of the subset from the superblock_hash
+            let a = u64::from(*self.current_superblock_hash.as_ref().get(0).unwrap());
+            let b = a % u64::from(_signing_committee_size);
+            let first_member = b % n;
+            // Get the subset
+            let subset = magic_partition(&ars_ordered.to_vec(), first_member, n);
+            let hs: HashSet<PublicKeyHash> = subset.iter().cloned().collect();
```

Comment:
This could be an alternative to the `magic_partition` thing:
```suggestion
            let subset: HashSet<PublicKeyHash> = ars_ordered.iter()
                .cloned()
                .enumerate()
                .filter_map(|(index, member)| {
                    if (index + offset) % possible_subsets == 0 {
                        Some(member)
                    } else {
                        None
                    }
                })
                .collect()
```
(Where `offset` was `a` and `possible_subsets` was `n` before)